File: Python Scripts/Bioinformatics/pointMutationCounting.py
#Counting Point Mutations

#The following script takes input from the user, in the form of two strings representing two different DNA strands.
#A function is then used to iterate through both strings and compare the nucleotide bases in them. If the two characters being compared
#do not match, the counter is incremented and the result printed to the console.

string_1 = input("Please enter a DNA strand:")

string_2 = input("Please enter another DNA strand:")

string_1_upper = string_1.upper()

string_2_upper = string_2.upper()
#function for comparing two strings and counting number of differing characters (Hamming Distance):
def p_m_loop(string_1, string_2):
	p_mutation = 0
	for position in range(len(string_1)):
		if string_1[position] != string_2[position]:
			p_mutation += 1
	return p_mutation
print(p_m_loop(string_1, string_2))
	

File: C# Code/Bioinformatics/pointMutationCounting.cs
/*Counting Point Mutations

The following script takes input from the user, in the form of two strings representing two different DNA strands.
A function is then used to iterate through both strings and compare the nucleotide bases in them. If the two characters being compared
do not match, the counter is incremented and the result printed to the console.
*/
using System;
using System.Collections.Generic;
using System.Text;
using System.Linq;
using pointMutationCounter;

namespace pointMutationCounter
{
    class pointMutCount
    {
        public static void Main(string[] args)
        {
            Console.WriteLine("Please enter a DNA strand: ");

            string string_1 = Console.ReadLine();

            Console.WriteLine("Please enter another DNA strand: ");

            string string_2 = Console.ReadLine();

            string_1 = string_1.ToUpper();

            string_2 = string_2.ToUpper();

            pointMutCount p = new pointMutCount();

            Console.WriteLine("The number of point mutations between the two DNA strands is:");

            Console.WriteLine(p.p_m_loop(string_1, string_2));

        }

        public string p_m_loop(string string_1, string string_2)
        {
            int p_mutation = 0;
            int i;
            int l = (string_1.Length - 1);

            for(i = 0; i <= l; i++)
            {
                if(string_1[i] != string_2[i])
                {
                    p_mutation++;
                }
            }
            return p_mutation.ToString();
            
        }
    }
}
